main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Commands/FileTabCommands.cs

```diff
@@ -92,15 +108,20 @@ protected override ViewContent GetDocumentException ()
 			return active == null ? null : active.Window.ViewContent;
 		}
 	}
-	
+
+	class CloseAllToTheRightHandler : CloseAllButThisHandler
+	{
+		protected override bool StartAfterException => true;
```

Comment:
What about enabling it only if there are any documents to the right?
We could add the same to `CloseAllButThisHandler` but without the `.SkipWhile()` part.

```suggestion
		protected override bool StartAfterException => true;
		
		
		protected override void Update (CommandInfo info)
		{
			var active = IdeApp.Workbench.ActiveDocument;
			if (active == null)
				return;

			var activeNotebook = ((SdiWorkspaceWindow)active.Window).TabControl;
			info.Enabled = IdeApp.Workbench.Documents.SkipWhile (doc => doc != active).Any (doc => doc != active && ((SdiWorkspaceWindow)doc.Window).TabControl == activeNotebook);
		}
```

Reply:
@sevoku I have mixed feelings about this, which is why I left it out. How often is `Update` called for menu items? I know it's a lot for always-visible toolbar commands.

VScode doesn't bother disabling for the right-most tab, for what it's worth.

Happy to add this if we're not worried about `Update` doing work.

Reply:
I think the update is only called when the popup is opened. You only need to check is if is the last element of the array.

Reply:
Last element of the array with matching `TabControl`, yeah. I'll update on Monday, thanks!